internal/search-history.go

```diff
@@ -0,0 +1,300 @@
+package internal
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"unicode"
+
+	log "github.com/sirupsen/logrus"
+)
+
+/*
+Search history semantics:
+- On startup, load history or import from other pager
+- Any change to the input box resets the history index to "past the end"
+- If the user starts typing, then arrows up once and down once, whatever the
+  user typed should still be there.
+- On importing from less, remove unprintable characters
+- On exiting search, no matter how, add a new entry at the end and deduplicate.
+  Save to disk.
+*/
+
+var searchHistory []string
+
+const maxSearchHistoryEntries = 100
```

Comment:
Might make sense to re-use `less`'s env var if exists I guess.
```
       LESSHISTSIZE
              The maximum number of commands to save in the history file.  The default is 100.
```

Reply:
I'll skip having this configurable for now. I bumped it to 640. If that's not enough for somebody I'll decide then whether to bump it or make it configurable.